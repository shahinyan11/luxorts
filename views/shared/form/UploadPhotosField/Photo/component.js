import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import InputField from 'views/shared/form/InputField';
import SvgIcon from 'views/shared/SvgIcon';

const Photo = ({ id, src, isMain, onRemove, onEdit, onPreview, descriptionPath }) => (
  <div id={id} className="upload-photos__item">
    <div
      onClick={onPreview}
      className="upload-photos__card"
      style={{ backgroundImage: `url(${src})` }}
      role="button"
    >
      <div className="upload-photos__card-actions d-flex justify-content-space-between">
        <div onClick={onEdit} className="upload-photos__card-action" role="button">
          <SvgIcon icon="edit" className="icon-left" />
        </div>
        <div onClick={onRemove} className="upload-photos__card-action" role="button">
          <SvgIcon icon="cross" className="icon-left" />
        </div>
      </div>
      {isMain && (
        <div className="upload-photos__card-main-label">
          <FormattedMessage id="shared.mainPhoto" />
        </div>
      )}
    </div>
    <InputField name={descriptionPath} placeholder={{ id: 'shared.addCaption' }} />
  </div>
);

Photo.propTypes = {
  id: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
  descriptionPath: PropTypes.string.isRequired,
  isMain: PropTypes.bool.isRequired,
  onRemove: PropTypes.func.isRequired,
  onEdit: PropTypes.func.isRequired,
  onPreview: PropTypes.func.isRequired,
};

export default Photo;
