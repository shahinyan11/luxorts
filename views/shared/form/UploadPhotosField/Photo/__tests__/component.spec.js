import { shallow } from 'enzyme';

import Photo from '../component';

describe('UploadPhotosField. Photo component', () => {
  const props = {
    id: 'id',
    src: 'src',
    isMain: false,
    onRemove: jest.fn(),
    onEdit: jest.fn(),
    onPreview: jest.fn(),
    descriptionPath: 'descriptionPath',
  };

  const component = shallow(<Photo {...props} />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when isMain equals true', () => {
    component.setProps({
      isMain: true,
    });

    expect(component).toMatchSnapshot();
  });
});
