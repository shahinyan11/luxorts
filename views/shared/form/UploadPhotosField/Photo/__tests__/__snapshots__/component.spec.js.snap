// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadPhotosField. Photo component matches snapshot 1`] = `
<div
  className="upload-photos__item"
  id="id"
>
  <div
    className="upload-photos__card"
    onClick={[MockFunction]}
    role="button"
    style={
      Object {
        "backgroundImage": "url(src)",
      }
    }
  >
    <div
      className="upload-photos__card-actions d-flex justify-content-space-between"
    >
      <div
        className="upload-photos__card-action"
        onClick={[MockFunction]}
        role="button"
      >
        <SvgIcon
          className="icon-left"
          icon="edit"
        />
      </div>
      <div
        className="upload-photos__card-action"
        onClick={[MockFunction]}
        role="button"
      >
        <SvgIcon
          className="icon-left"
          icon="cross"
        />
      </div>
    </div>
  </div>
  <InputField
    asComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "Group": [Function],
        "Password": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "action": "click",
            "iconRender": [Function],
            "visibilityToggle": true,
          },
          "render": [Function],
        },
        "Search": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "TextArea": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "render": [Function],
      }
    }
    name="descriptionPath"
    numbersOnly={false}
    placeholder={
      Object {
        "id": "shared.addCaption",
      }
    }
    withFormItem={true}
  />
</div>
`;

exports[`UploadPhotosField. Photo component matches snapshot when isMain equals true 1`] = `
<div
  className="upload-photos__item"
  id="id"
>
  <div
    className="upload-photos__card"
    onClick={[MockFunction]}
    role="button"
    style={
      Object {
        "backgroundImage": "url(src)",
      }
    }
  >
    <div
      className="upload-photos__card-actions d-flex justify-content-space-between"
    >
      <div
        className="upload-photos__card-action"
        onClick={[MockFunction]}
        role="button"
      >
        <SvgIcon
          className="icon-left"
          icon="edit"
        />
      </div>
      <div
        className="upload-photos__card-action"
        onClick={[MockFunction]}
        role="button"
      >
        <SvgIcon
          className="icon-left"
          icon="cross"
        />
      </div>
    </div>
    <div
      className="upload-photos__card-main-label"
    >
      <Memo(FormattedMessage)
        id="shared.mainPhoto"
      />
    </div>
  </div>
  <InputField
    asComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "Group": [Function],
        "Password": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "action": "click",
            "iconRender": [Function],
            "visibilityToggle": true,
          },
          "render": [Function],
        },
        "Search": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "TextArea": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "render": [Function],
      }
    }
    name="descriptionPath"
    numbersOnly={false}
    placeholder={
      Object {
        "id": "shared.addCaption",
      }
    }
    withFormItem={true}
  />
</div>
`;
